jenkins/ompi/ompi_test.sh

```diff
@@ -110,265 +103,226 @@ mlnx_cov="SKIP"
-    if [ "$1" = "--no-bind" ]; then
+
+    if [ "$1" = "--no-bind" ]
+    then
         AFFINITY=""
         shift
     fi
 
-    local np=$1
+    local np="$1"
     local exe_path="$2"
-    local exe_args=${3}
-    local common_mca="--bind-to none --mca orte_tmpdir_base $jenkins_session_base"
-    local mpirun="$OMPI_HOME/bin/mpirun"
-
-    local has_timeout=$($OMPI_HOME/bin/mpirun --help | grep timeout | wc -l)
-    if [ "$has_timeout" = "0" ]; then
+    local exe_args="$3"
+    local common_mca="--bind-to none --mca orte_tmpdir_base ${ci_session_base}"
+    local mpirun="${OMPI_HOME}/bin/mpirun"
+    local has_timeout
+    has_timeout=$("${OMPI_HOME}/bin/mpirun" --help | grep --count timeout || true)
+
+    if [ "${has_timeout}" = "0" ]
+    then
         # Help in newer versions was changed
-        has_timeout=$($OMPI_HOME/bin/mpirun --help debug 2>/dev/null | grep timeout | wc -l)
+        has_timeout=$("${OMPI_HOME}/bin/mpirun" --help debug 2>/dev/null | grep --count timeout || true)
     fi
-    if [ "$has_timeout" -gt 0 ]; then
+
+    if [ "${has_timeout}" -gt 0 ]
+    then
         # We leave external timeout just in case the internal one is broken
-        common_mca="$common_mca $mpi_timeout"
+        common_mca="${common_mca} ${mpi_timeout}"
     fi
 
-    if [ "$jenkins_test_hcoll" = "no" ]; then
-        common_mca="$common_mca --mca coll ^hcoll"
+    if [ "${ci_test_hcoll}" = "no" ]
+    then
+        common_mca="${common_mca} --mca coll ^hcoll"
     fi
-    local mca="$common_mca"
 
-    if [ "$btl_tcp" == "yes" ]; then
-        $timeout_exe $mpirun --np $np $mca --mca pml ob1 --mca btl self,tcp ${AFFINITY} ${exe_path} ${exe_args}
-    fi
+    local mca="${common_mca}"
 
-    if [ "$btl_sm" == "yes" ]; then
-        $timeout_exe $mpirun --np $np $mca --mca pml ob1 --mca btl self,sm ${AFFINITY} ${exe_path} ${exe_args}
+    if [ "${btl_tcp}" = "yes" ]
+    then
+        # shellcheck disable=SC2086
+        ${timeout_exe} "$mpirun" --np "$np" $mca --mca pml ob1 --mca btl self,tcp ${AFFINITY} "${exe_path}" "${exe_args}"
     fi
 
-    if [ "$btl_vader" == "yes" ]; then
-        $timeout_exe $mpirun --np $np $mca --mca pml ob1 --mca btl self,vader ${AFFINITY} ${exe_path} ${exe_args}
+    if [ "${btl_sm}" = "yes" ]
+    then
+        # shellcheck disable=SC2086
+        ${timeout_exe} "$mpirun" --np "$np" $mca --mca pml ob1 --mca btl self,sm ${AFFINITY} "${exe_path}" "${exe_args}"
     fi
 
+    if [ "${btl_vader}" = "yes" ]
+    then
+        # shellcheck disable=SC2086
+        ${timeout_exe} "$mpirun" --np "$np" $mca --mca pml ob1 --mca btl self,vader ${AFFINITY} "${exe_path}" "${exe_args}"
+    fi
 
-    local has_yalla=$($OMPI_HOME/bin/ompi_info --param pml all --level 9 | grep yalla | wc -l)
-    local has_ucx=$($OMPI_HOME/bin/ompi_info --param pml all --level 9 | grep ucx | wc -l)
-    local has_btl_openib=$($OMPI_HOME/bin/ompi_info --param btl all | grep openib | wc -l)
-    for hca_dev in $(ibstat -l); do
+    local has_ucx
```

Comment:
refactor block to run always with `ucx`, all other remove

Reply:
Fixed.